### PR TITLE
Tests of the 1D ODE solvers provided by the gauss_legendre.jl module.…

### PR DESCRIPTION
**Pull request by mrhardman**
… These 1D ODE solvers are useful to show the basics of how the more complex 2D PDE solvers are constructed, and to show how different types of boundary conditions are imposed.

@johnomotani These tests make sure that the 1D ODE solvers are not broken in future commits. I hope that this PR is sufficiently simple to be merged quickly.

### CODE REVIEW

**Comment by mrhardman** [comment]
@johnomotani Unclear why the parallel Ubuntu job failed here. Is this related to #259 ?

**Comment by johnomotani** [comment]
> Unclear why the parallel Ubuntu job failed here. Is this related to #259?

Don't know, but could well be. I've been having problems with the parallel CI job, see 3cd263e and e7a908c (which were merged with #247) as well as #259.

**Comment by mrhardman** [comment]
I guess you reran the test and it passed? Can I merge this now?

File: moment_kinetics/test/calculus_tests.jl [hunk 1]
```diff
@@ -1565,6 +1566,297 @@ function runtests()
...
+                                                     element_spacing_option=element_spacing_option,
+                                                     collision_operator_dim=false)
+                expected_f = @. exp(-x.grid^2)
+                d2f = @. 4.0*(x.grid^2 - 1.0)*exp(-x.grid^2)
+                # create array for the numerical solution
+                f = similar(expected_f)
+                # create array for RHS vector b
+                b = similar(expected_f)
+                # solve for f
+                mul!(b,spectral.mass_matrix,d2f)
```
Comment (johnomotani) [inline moment_kinetics/test/calculus_tests.jl:1660]: `d2f` in other places would mean `d^2f/dx^2`, so I think it's clearer to use a different variable name. Also could use a comment on what equation is solved
```suggestion
                # Test solver for
                #   Laplacian_f = 1/x * d/dx(x*df/dx)
                Laplacian_f = @. 4.0*(x.grid^2 - 1.0)*exp(-x.grid^2)
                # create array for the numerical solution
                f = similar(expected_f)
                # create array for RHS vector b
                b = similar(expected_f)
                # solve for f
                mul!(b,spectral.mass_matrix,Laplacian_f)
```

File: moment_kinetics/test/calculus_tests.jl [hunk 2]
```diff
@@ -1565,6 +1566,297 @@ function runtests()
...
+                # `ignore_MPI=true` to avoid errors due to communicators not being fully
+                # set up.
+                x, spectral = define_test_coordinate("vpa"; ngrid=ngrid,
+                                                     nelement=nelement, L=L,
+                                                     discretization="gausslegendre_pseudospectral",
+                                                     bc=bc,
+                                                     element_spacing_option=element_spacing_option,
+                                                     collision_operator_dim=false)
+                expected_f = @. exp(-x.grid^2)
+                d2f = @. (4.0*x.grid^2 - 2.0)*exp(-x.grid^2)
```
Comment (johnomotani) [inline moment_kinetics/test/calculus_tests.jl:1745]:
```suggestion
                # Test solver for
                #   d2f = d^2f/dx^2
                d2f = @. (4.0*x.grid^2 - 2.0)*exp(-x.grid^2)
```

File: moment_kinetics/test/calculus_tests.jl [hunk 3]
```diff
@@ -1565,6 +1566,297 @@ function runtests()
...
+                # `ignore_MPI=true` to avoid errors due to communicators not being fully
+                # set up.
+                x, spectral = define_test_coordinate("vpa"; ngrid=ngrid,
+                                                     nelement=nelement, L=L,
+                                                     discretization="gausslegendre_pseudospectral",
+                                                     bc=bc,
+                                                     element_spacing_option=element_spacing_option,
+                                                     collision_operator_dim=false)
+                expected_f = @. sin((2.0*pi*x.grid/x.L) + phase)
+                d2f = @. -((2.0*pi/x.L)^2)*sin((2.0*pi*x.grid/x.L)+phase)
```
Comment (johnomotani) [inline moment_kinetics/test/calculus_tests.jl:1839]:
```suggestion
                # Test solver for
                #   d2f = d^2f/dx^2
                d2f = @. -((2.0*pi/x.L)^2)*sin((2.0*pi*x.grid/x.L)+phase)
```

**Review by johnomotani — APPROVED** [review]
